neon-tracer/src/data_source/neon_api.rs

```
@@ -116,46 +110,35 @@ impl NeonAPIDataSource {
     }
 
     pub async fn get_balance(&self, get_ether_request: GetEtherRequest, id: u64) -> Result<U256> {
-        let response = self
+        match self
             .api_client
             .get_ether_account_data(get_ether_request, id)
-            .await;
-
-        if response.is_err() {
-            Ok(U256::default())
-        } else {
-            U256::from_str_prefixed(&response.unwrap().balance)
-                .map_err(|_| ERR("cast GetEtherAccountDataReturn.balance->U256", id))
+            .await
+        {
+            Err(_) => Ok(U256::default()),
+            Ok(r) => U256::from_str_prefixed(&r.balance)
+                .context("Invalid balance returned by neon_api::get_ether_account_data"),
         }
     }
 
-    pub async fn get_trx_count(&self, get_ether_request: GetEtherRequest, id: u64) -> Result<U256> {
-        let response = self
-            .api_client
+    pub async fn get_trx_count(&self, get_ether_request: GetEtherRequest, id: u64) -> U256 {
+        self.api_client
             .get_ether_account_data(get_ether_request, id)
-            .await;
-
-        if response.is_err() {
-            Ok(U256::default())
-        } else {
-            Ok(U256::new(response.unwrap().trx_count.into()))
-        }
+            .await
+            .map(|r| U256::new(r.trx_count.into()))
+            .unwrap_or_default()
     }
 
-    pub async fn get_code(&self, get_ether_request: GetEtherRequest, id: u64) -> Result<String> {
-        let response = self
-            .api_client
+    pub async fn get_code(&self, get_ether_request: GetEtherRequest, id: u64) -> String {
+        self.api_client
             .get_ether_account_data(get_ether_request, id)
-            .await;
-
-        if response.is_err() {
-            Ok(String::default())
-        } else {
-            Ok(response.unwrap().code)
-        }
+            .await
+            .map(|r| r.code)
+            .unwrap_or_default()
```

Let's bubble up the error instead of shadowing it into a default value.